android/app/src/main/java/org/uwaterloo/subletr/pages/createlisting/CreateListingPageViewModel.kt

```diff
@@ -16,27 +27,62 @@ import org.uwaterloo.subletr.api.apis.ListingsApi
 import org.uwaterloo.subletr.api.models.CreateListingRequest
 import org.uwaterloo.subletr.api.models.ListingsImagesCreateRequest
 import org.uwaterloo.subletr.api.models.ResidenceType
-import org.uwaterloo.subletr.enums.Gender
 import org.uwaterloo.subletr.enums.HousingType
 import org.uwaterloo.subletr.infrastructure.SubletrViewModel
-import org.uwaterloo.subletr.services.ILocationService
 import org.uwaterloo.subletr.services.INavigationService
+import org.uwaterloo.subletr.utils.CANADA
+import org.uwaterloo.subletr.utils.UWATERLOO_LATITUDE_DOUBLE
+import org.uwaterloo.subletr.utils.UWATERLOO_LONGITUDE_DOUBLE
 import org.uwaterloo.subletr.utils.toBase64String
+import java.util.concurrent.TimeUnit
 import javax.inject.Inject
 
 @HiltViewModel
 class CreateListingPageViewModel @Inject constructor(
 	private val listingsApi: ListingsApi,
 	val navigationService: INavigationService,
+	@ApplicationContext applicationContext: Context,
 ) : SubletrViewModel<CreateListingPageUiState>() {
 	val navHostController: NavHostController get() = navigationService.navHostController
+	private val autocompleteSessionToken: AutocompleteSessionToken = AutocompleteSessionToken.newInstance()
+	private val bounds: RectangularBounds = RectangularBounds.newInstance(
+		LatLng(UWATERLOO_LATITUDE_DOUBLE, UWATERLOO_LONGITUDE_DOUBLE),
+		LatLng(UWATERLOO_LATITUDE_DOUBLE, UWATERLOO_LONGITUDE_DOUBLE)
+	)
+	private val placesClient: PlacesClient = Places.createClient(applicationContext)
 
 	val fullAddressStream: BehaviorSubject<String> = BehaviorSubject.createDefault("")
+	private val debouncedAddressStream: Observable<String> = fullAddressStream
+		.debounce(1, TimeUnit.SECONDS, Schedulers.computation())
+	private val addressAutocompleteStream: Observable<ArrayList<AutocompletePrediction>> = debouncedAddressStream
```

Comment:
it works and I don't see performance issues but if something's not done right... I'll fix it

Reply:
combined